packages/noya-state/src/reducers/application.ts

```diff
@@ -1257,6 +1277,231 @@ export function reducer(
         else draft.selectedThemeStyleGroup = '';
       });
     }
+    case 'setTextColor': {
+      const [, color] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded = layer.style?.textStyle?.encodedAttributes;
+          const attributes = layer.attributedString.attributes[0].attributes;
+
+          if (encoded) {
```

Comment:
Maybe both the `encoded` and `attributes` case here could call a `changeColor` function or something, if they're similar.